File: frontend/messenger/src/types/ConversationDTO.ts
import { EmojiDTO } from "./EmojiDTO";
import { ThemeDTO } from "./ThemeDTO";

export interface ConversationDTO {
    id: number;
    type: string;
    title: string;
    imageData: Uint8Array;
    theme: ThemeDTO;
    emoji: EmojiDTO;
    createdAt: string;
    updatedAt: string;
}

File: backend/messenger/src/main/java/com/example/messenger/chat/conversation/DTO/ConversationDTO.java
package com.example.messenger.chat.conversation.DTO;

import com.example.messenger.chat.conversation.emoji.model.Emoji;
import com.example.messenger.chat.conversation.model.ConversationType;
import com.example.messenger.chat.conversation.theme.model.Theme;

import java.time.LocalDateTime;

public record ConversationDTO(
        Long id,
        ConversationType type,
        String title,
        byte[] imageData,
        Theme theme,
        Emoji emoji,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {
}
